queries.sql

```diff
@@ -1,3 +1,244 @@
+
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+select owners.full_name, animals.name , animals.escape_attempts from animals inner join owners
+on owners.id=animals.owner_id
+where owners.full_name='Dean Winchester'
+and animals.escape_attempts=0;
+select * from animals
+
+-- Who owns the most animals?
+
+select  count(animals.name) total_animals, owners.full_name from animals inner join owners
+on owners.id= animals.owner_id
+group by owners.full_name
+order by total_animals desc
+limit 1
+
+
+
+-- queries for many to many relations
+
+-- Who was the last animal seen by William Tatcher?
+
+select animals.name, vets.name, visits.date_of_vists from animals inner join visits on animals.id=visits.animal_id
+inner join vets on visits.vets_id=vets.id
+where vets.name='William Tatcher'
+order by visits.date_of_vists desc
+limit 1
+
+-- How many different animals did Stephanie Mendez see?
+
+select count(animals.name), vets.name from animals inner join visits on animals.id=visits.animal_id
+inner join vets on visits.vets_id=vets.id
+where vets.name='Stephanie Mendez'
+group by vets.name
+
+
+-- List all vets and their specialties, including vets with no specialties.
+
+select vets.name, species.name
+from specializations inner join species on specializations.species_id = species.id
+right outer join vets on specializations.vets_id = vets.id
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+select animals.name, vets.name, visits.date_of_vists from animals inner join visits on animals.id=visits.animal_id
+inner join vets on visits.vets_id=vets.id
+where vets.name='Stephanie Mendez'
+and visits.date_of_vists BETWEEN '01-04-2020' AND '08-30-2020'
+
+-- What animal has the most visits to vets?
+Select count(animals.name), vets.name from animals inner join visits on animals.id=visits.animal_id
+inner join vets on vets.id= visits.vets_id
+group by vets.name
+order by count(animals.name) desc
+limit 1
+
+-- Who was Maisy Smith's first visit?
+
+Select animals.name, vets.name , visits.date_of_vists from animals inner join visits on animals.id=visits.animal_id
+inner join vets on vets.id= visits.vets_id
+where vets.name='Maisy Smith'
+order by visits.date_of_vists 
+limit 1
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+
+
+Select animals.name, vets.name , visits.date_of_vists from animals inner join visits on animals.id=visits.animal_id
+inner join vets on vets.id= visits.vets_id
+order by visits.date_of_vists desc
+limit 1
+
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+
+SELECT COUNT(visits.date_of_vists)-(SELECT COUNT(vets.name) 
+FROM vets, specializations, animals, visits
+WHERE
+visits.vets_id = vets.id
+AND animals.id = visits.animal_id
+AND concat(animals.species_id, visits.vets_id) = concat(specializations.species_id, specializations.vets_id)) total_unspecialized_cases 
+FROM visits;
```

Comment:
- [ ] only Maisy Smith doesn't specialize in a species and she has 8 visits to her name, and the result of this query is 13.

Reply:
@mikethreels 

this is the query

  SELECT animals.name, visits.date_of_vists, vets.name, visits.vets_id, animals.species_id
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vets_id
  INNER JOIN species ON species.id = animals.species_id


![image](https://user-images.githubusercontent.com/107524451/222921119-bea3fc60-8468-4bff-9484-485875eae67d.png)
![image](https://user-images.githubusercontent.com/107524451/222921131-c585b1d7-9a62-4742-abb5-4a1395ddc6f2.png)

I hope you are doing well. I have taken the screenshot and highlighted those visits which were non-specialized vets visits. Kindly count them. It is not just about Maisy smith Actually other vets who are not specialized in other species they have done visits to their animals too.